pkg/service/service.go

```diff
@@ -0,0 +1,196 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/kardianos/service"
+	"github.com/spf13/cobra"
+)
+
+// Service is an application that runs supervised by the OS
+type Service struct {
+	wg         sync.WaitGroup
+	cancelOnce sync.Once
+
+	cancel context.CancelFunc
+	ctx    context.Context
+
+	d   time.Duration
+	err atomic.Value
+	ss  service.Service
+	p   program
+
+	cmd     cobra.Command
+	prepare func(context.Context, *cobra.Command, []string) error
+	run     func(context.Context, *cobra.Command, []string) error
+
+	initializer []func()
```

Comment:
Why is this singular and `finalizers` plural? seem sto me that this should be `initializers`

Reply:
good catch, thanks